### ISSUE TITLE
Replacement for ast_to_dict?

### ISSUE THREAD

**Issue by ptrhck** [graphql-python/graphql-core#136]
I have been using [this](https://gist.github.com/mixxorz/dc36e180d1888629cf33#file-graphene-py-L59) function in the past, but it seems that `ast_to_dict` has been removed. Is there a replacement for `ast_to_dict` or a new function that is similiar to the one of the gist?

**Comment by Cito** [comment]
That function `ast_to_dict` does not exist in GraphQL.js, but [was added](https://github.com/graphql-python/graphql-core-legacy/commit/fd83a778f173cef4502073a5cdfe7cd7526bec95) to the legacy version of GraphQL-core because it makes sense to have this (only) in Python.

So yes, might be a good idea to add this to the modern GraphQL-core as well. The function could look like this, you can use this as a workaround for the time being:

```python
def ast_to_dict(node, with_location=False):
    if isinstance(node, Node):
        res = {key: ast_to_dict(getattr(node, key), with_location)
               for key in node.keys if key != 'loc'}
        if with_location:
            loc = node.loc
            if loc:
                res['loc'] = dict(start=loc.start, end=loc.end)
        res['kind'] = node.kind
        return res
    if isinstance(node, list):
        return [ast_to_dict(item, with_location) for item in node]
    return node
```

Btw, an `ast_to_code` function [was also added](https://github.com/graphql-python/graphql-core-legacy/pull/12/commits/889fc343d3d5b17e8b3edbc49af60c99d8102b89), we should consider adding this as well.

**Comment by Cito** [comment]
Added this back now, will be available in v3.2.
